src/main/java/org/springframework/data/tarantool/core/convert/MappingTarantoolReadConverter.java

```diff
@@ -101,7 +101,13 @@ public <R> R read(Class<R> targetClass, final @Nullable Object source) {
 
         TypeInformation<? extends R> typeToUse;
         if (source instanceof TarantoolTuple) {
-            typeToUse = typeMapper.readType((TarantoolTuple) source, ClassTypeInformation.from(targetClass));
+            TarantoolTuple tuple = (TarantoolTuple) source;
+            if (tuple.hasMetadata()) {
```

Comment:
```suggestion
            if (!tuple.hasMetadata()) {
```

Reply:
We need a test for that part, we should have caught this error

Reply:
Yeah, I guess it was wrong to return a true if metadata is null. I don't know how this is happened 🙃
```java
@Override
public boolean hasMetadata() {
    return spaceMetadata == null || spaceMetadata.getSpaceFormatMetadata() == null ||
               spaceMetadata.getSpaceFormatMetadata().isEmpty();
}
```
https://github.com/tarantool/cartridge-java/blob/d9ca853b6008137c2ea6c68a0246a68d43771d31/src/main/java/io/tarantool/driver/core/tuple/TarantoolTupleImpl.java#L140-L144

Reply:
I got how it happened. First we have `metadataFormatIsEmpty`, you suggest renaming it to `hasMetadata`. I renamed it but I didn't think about changing logic, because it's opposite behaviour by names